bin/reth/src/cli/mod.rs

```diff
@@ -245,6 +233,22 @@ pub enum Commands<C: ChainSpecParser, Ext: clap::Args + fmt::Debug> {
     Prune(prune::PruneCommand<C>),
 }
 
+/// A provider for chain specifications.
+pub trait ChainSpecProvider<C: ChainSpecParser> {
+    /// Returns an optional chain specification.
+    fn chain_spec(&self) -> Option<&C::ChainSpec>;
+}
+
+impl<C: ChainSpecParser, Ext: clap::Args + fmt::Debug> ChainSpecProvider<C> for Commands<C, Ext> {
+    fn chain_spec(&self) -> Option<&C::ChainSpec> {
+        match self {
+            Commands::Node(cmd) => Some(&cmd.chain),
+            // For other subcommands that do not provide a chain, return None.
+            _ => None,
+        }
+    }
+}
```

Comment:
I'd rather convert this to a native function of `Commands`

Reply:
Makes sense, done.